src/app/task/create-task/create-task.component.ts

```diff
@@ -28,8 +28,15 @@ export class CreateTaskComponent {
   constructor(private taskService: TaskService, private router: Router) {}
 
   createTask() {
-    this.newTask = new Task(this.taskForm.value.title, this.taskForm.value.description, this.taskForm.value.type);
-    this.taskService.createTask(this.newTask);
-    this.router.navigate(['/tasks/task-list']);
+    if (this.taskForm.value.title && this.taskForm.value.description && this.taskForm.value.type) {
+      this.newTask = new Task(this.taskForm.value.title, this.taskForm.value.description, this.taskForm.value.type);
```

Comment:
No need to create a variable outside of the method, instead you can define it inside the method like
let newTask = new Task(etc, etc, etc)